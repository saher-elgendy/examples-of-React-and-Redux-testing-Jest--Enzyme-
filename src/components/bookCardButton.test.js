import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import BookCardbutton from './bookCardButton';
import './../setupTests';


describe('BookStore component', () => {
    let wrapper;
    let mockProps;

    beforeEach(() => {
        mockProps = {
            bag: [],
            book: {},
            addBookToBag: jest.fn(),
            removeBookFromBag: jest.fn()
        }
        wrapper = shallow(<BookCardbutton {...mockProps} />)
    })

    it('render BookStore component correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })
})