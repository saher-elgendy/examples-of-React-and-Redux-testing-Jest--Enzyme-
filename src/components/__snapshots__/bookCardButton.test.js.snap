// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookStore component render BookStore component correctly 1`] = `
<button
  className="book-card-btn add-book-btn"
  onClick={[Function]}
>
  ADD TO BAG
</button>
`;
